### PR TITLE
New patch for gsl with external-cblas

### PR DESCRIPTION
**Pull request by glennpj**
The wavelet hunk changed with version 2.6.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gsl/package.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ class Gsl(AutotoolsPackage, GNUMirrorPackage):
     variant('external-cblas', default=False, description='Build against external blas')
 
     # from https://dev.gentoo.org/~mgorny/dist/gsl-2.3-cblas.patch.bz2
-    patch('gsl-2.3-cblas.patch', when="+external-cblas")
+    patch('gsl-2.3-cblas.patch', when="@:2.5 +external-cblas")
+    patch('gsl-2.6-cblas.patch', when="@2.6: +external-cblas")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/gsl/package.py:33]: Did you create this patch yourself or does it come from somewhere?

Comment (glennpj) [reply var/spack/repos/builtin/packages/gsl/package.py:33]: This is a copy of the `gsl-2.3-cblas.patch` file, with a change to the wavelet hunk.
```
diff -u var/spack/repos/builtin/packages/gsl/gsl-2.*.patch
--- var/spack/repos/builtin/packages/gsl/gsl-2.3-cblas.patch    2020-08-20 15:27:46.433233535 -0500
+++ var/spack/repos/builtin/packages/gsl/gsl-2.6-cblas.patch    2021-04-13 16:33:42.023213171 -0500
@@ -316,8 +316,8 @@
  
  TESTS = $(check_PROGRAMS)
  
--test_LDADD = libgslwavelet.la ../blas/libgslblas.la ../cblas/libgslcblas.la ../matrix/libgslmatrix.la ../vector/libgslvector.la ../block/libgslblock.la ../ieee-utils/libgslieeeutils.la  ../err/libgslerr.la ../test/libgsltest.la ../sys/libgslsys.la ../utils/libutils.la
-+test_LDADD = libgslwavelet.la ../blas/libgslblas.la @CBLAS_LINK_LIBS@ ../matrix/libgslmatrix.la ../vector/libgslvector.la ../block/libgslblock.la ../ieee-utils/libgslieeeutils.la  ../err/libgslerr.la ../test/libgsltest.la ../sys/libgslsys.la ../utils/libutils.la
+-test_LDADD = libgslwavelet.la ../matrix/libgslmatrix.la ../vector/libgslvector.la ../blas/libgslblas.la ../cblas/libgslcblas.la ../block/libgslblock.la ../ieee-utils/libgslieeeutils.la  ../err/libgslerr.la ../test/libgsltest.la ../sys/libgslsys.la ../utils/libutils.la
++test_LDADD = libgslwavelet.la ../matrix/libgslmatrix.la ../vector/libgslvector.la ../blas/libgslblas.la @CBLAS_LINK_LIBS@ ../block/libgslblock.la ../ieee-utils/libgslieeeutils.la  ../err/libgslerr.la ../test/libgsltest.la ../sys/libgslsys.la ../utils/libutils.la
  
  test_SOURCES = test.c
```
So, the source for me was the existing patch.

Comment (glennpj) [reply var/spack/repos/builtin/packages/gsl/package.py:33]: @scheibelp Does this look okay to be merged?

**Comment by glennpj** [comment]
Can someone please take a look at this? This patch is needed for the external cblas to be linked to the gsl-2.6 library. Dropping the patch for gsl-2.6, as was done in PR #23787, is incorrect.

**Review by scheibelp — APPROVED** [review]

File: var/spack/repos/builtin/packages/gsl/package.py (hunk 1 above)
Comment (scheibelp) [reply var/spack/repos/builtin/packages/gsl/package.py:33]: Yes - thanks and sorry for taking so long to check back on this!

**Pull request merged**